### PR TITLE
Add an integration test for blkio statistics.

### PR DESCRIPTION
**Pull request by qianzhangxa**
## High Level Description

Add an integration test to make sure blkio statistics can be successfully collected from Mesos agent's endpoint `/monitor/statistics`.

## Related Issues

  - [CORE-1235](https://jira.mesosphere.com/browse/CORE-1235) DC/OS integration tests for blkio stats.

## Checklist for all PR's

  - [ ] Included a test which will fail if code is reverted but test is not. If there is no test please explain here:
  - [x] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [x] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [ ] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae)
  - [ ] Test Results: [link to CI job test results for component]
  - [ ] Code Coverage (if available): [link to code coverage report]
___
**PLEASE FILL IN THE TEMPLATE ABOVE** / **DO NOT REMOVE ANY SECTIONS ABOVE THIS LINE**

## Instructions and Review Process

**What is the review process and when will my changes land?**

All PRs require 2 approvals using github's [pull request reviews](https://help.github.com/articles/about-pull-request-reviews/).

Reviewers should be:
* Developers who understand the code being modified.
* Developers responsible for code that interacts with or depends on the code being modified.

It is best to proactively ask for 2 reviews by @mentioning the candidate reviewers in the PR comments area. The responsibility is on the developer submitting the PR to follow-up with reviewers and make sure a PR is reviewed in a timely manner. Once a PR has **2 ship-it's**, **no red reviews**, and **all tests are green** it will be included in the [next train](https://github.com/dcos/dcos/blob/master/contributing.md).

### CODE REVIEW

**Comment by qianzhangxa** [comment]
@mesosphere-mergebot label Ready For Review

**Comment by qianzhangxa** [comment]
cc @vinodkone

**Comment by jgehrcke** [comment]
@mesosphere-mergebot bump-ee

**Comment by qianzhangxa** [comment]
@jgehrcke Thanks for your comments! I have updated the commit by adding more comments in the code, can you please take a look again?

File: packages/dcos-integration-test/extra/test_mesos.py [hunk 1]
```diff
@@ -171,6 +171,100 @@ def get_ps_output():
...
+        stats = r.json()
+
+        total_io_serviced = None
+        total_io_service_bytes = None
+        for stat in stats:
+            # Find the statistic for the Marathon application that we deployed. Since what that
+            # Marathon application launched is a Mesos command task (i.e., using Mesos built-in
+            # command executor), the executor ID will be same as the task ID, so if we find the
+            # `app_id` in an executor ID of a statistic, that must be the statistic entry
+            # corresponding to the application that we deployed.
```
Comment (jgehrcke) [inline packages/dcos-integration-test/extra/test_mesos.py]: Love that comment; thanks!!

File: packages/dcos-integration-test/extra/test_mesos.py [hunk 2]
```diff
@@ -171,6 +171,100 @@ def get_ps_output():
...
+        assert total_io_serviced is not None, ('Total blkio throttling IO serviced not found '
+                                               'for app_id {}'.format(app_id))
+        assert total_io_service_bytes is not None, ('Total blkio throttling IO service bytes '
+                                                    'not found for app_id {}'.format(app_id))
+        # We expect the statistics retrieved from Mesos agent are equal or greater than what we
+        # did with the `dd` command (i.e., 10 and 10240), because:
+        #   1. Besides the disk writes done by the `dd` command, the statistics may also include
+        #      some disk reads, e.g., to load the necessary executable binary and libraries.
+        #   2. In the environment where RAID is enabled, there may be multiple disk writes to
+        #      different disks for a single `dd` write.
```
Comment (jgehrcke) [inline packages/dcos-integration-test/extra/test_mesos.py]: Did we confirm that this does _not_ pass when we _don't_ do `dd if=/dev/zero of=file bs=1024 count=10 oflag=dsync`?

Comment (qianzhangxa) [reply packages/dcos-integration-test/extra/test_mesos.py]: Yeah, I have confirmed it. If I remove `dd if=/dev/zero of=file bs=1024 count=10 oflag=dsync`, the test will fail:
```
>           assert int(total_io_serviced) >= 10, ('Total blkio throttling IO serviced for app_id {} '
                                                  'are less than 10'.format(app_id))
E           AssertionError: Total blkio throttling IO serviced for app_id integration-test-a07ed659cd184ccc87a99635615f6fbc are less than 10
E           assert 1 >= 10
E            +  where 1 = int(1)

test_mesos.py:261: AssertionError
```

**Comment by qianzhangxa** [comment]
@jgehrcke Thanks for your comments! I have addressed all of them, can we merge this PR?

**Comment by jgehrcke** [comment]
I've asked @gpaul to make another review. @qianzhangxa in general you should simply ask others directly for getting quicker reviews :-).

**Comment by qianzhangxa** [comment]
Thanks @jgehrcke! This is my first time to submit a PR for DC/OS integration test, I just did not know whom I should ask to review the PR, will do it directly next time :-)

**Comment by jgehrcke** [comment]
@qianzhangxa there is a common misunderstanding in that area. "We" have tried to resolve that with the following part of the pull request template which you have surely read :-)

> All PRs require 2 approvals using github's pull request reviews.
> 
> Reviewers should be:
>
> - Developers who understand the code being modified.
> - Developers responsible for code that interacts with or depends on the code being modified.
> 
> It is best to proactively ask for 2 reviews by @mentioning the candidate reviewers in the PR comments area. The responsibility is on the developer submitting the PR to follow-up with reviewers and make sure a PR is reviewed in a timely manner.

No worries, but we all need to be a bit more proactive in that regard.

**Comment by vinodkone** [comment]
@orsenthil  Do I take that this will automatically get merged in the next train?

**Comment by jgehrcke** [comment]
> @orsenthil Do I take that this will automatically get merged in the next train?

This will only be put on the next train after having set the `ship-it` label.

**Comment by jgehrcke** [comment]
@mesosphere-mergebot bump-ee

**Comment by jgehrcke** [comment]
@vinodkone @qianzhangxa I am pushing this further ahead now, but please make sure your team can provide the same in the future :-).

**Comment by jgehrcke** [comment]
@mesosphere-mergebot ship_it_label_status_check

**Comment by vinodkone** [comment]
Thanks @jgehrcke ! Am I supposed to be doing "@mesosphere-mergebot ship_it_label_status_check
" after it got approvals?

**Comment by orsenthil** [comment]
@vinodkone  - Not required. A metronome job automatically applies in 5 minute intervals.

**Pull request merged**